syncer/ddl.go

```diff
@@ -33,25 +40,22 @@ type parseDDLResult struct {
 	isDDL  bool
 }
 
-func (s *Syncer) parseDDLSQL(sql string, p *parser.Parser, schema string) (result parseDDLResult, err error) {
-	sql = utils.TrimCtrlChars(sql)
+type shardingDDLInfo struct {
+	name       string
+	tableNames [][]*filter.Table
+	stmt       ast.StmtNode
+}
 
-	// check skip before parse (used to skip some un-supported DDLs)
-	ignore, err := s.skipQuery(nil, nil, sql)
-	if err != nil {
-		return parseDDLResult{
-			stmt:   nil,
-			ignore: false,
-			isDDL:  false,
-		}, err
-	} else if ignore {
+// filterDDLSQL filter two type of sql: 1. tidb not support 2. not ddl sql such as begin/flush.
```

Comment:
this function wants to filter sql of "tidb not support", and we provided `IsBuildInSkipDDL` to achieve that. But we might miss some pattern and need user to specify a regular expression to filter it.

so `ignore, err := s.skipQuery(nil, nil, originSQL)` should be called before any parsing.

We might add a test in CI, mariaDB has lots of unsupported SQL, we could execute one of them and check the binlog filter could work.